var/spack/repos/builtin/packages/direnv/package.py

```diff
@@ -15,7 +18,40 @@ class Direnv(Package):
     version('2.20.0', sha256='cc72525b0a5b3c2ab9a52a3696e95562913cd431f923bcc967591e75b7541bff')
     version('2.11.3', sha256='2d34103a7f9645059270763a0cfe82085f6d9fe61b2a85aca558689df0e7b006')
 
-    depends_on('go', type='build')
+    depends_on('go@1.11:', type='build')  # module/vendor support
```

Comment:
What actually prevents us from using the GoPackage base class for direnv? Does GoPackage need to be more flexible?

Reply:
Direnv is a mess.  It hasn't been updated in a while, uses a Makefile, and use[sd] an one of the community experiments for dependency management.  In spite of all that, it *does* have a vendor dir that contains everything we need (the Go modules design intentionally made the contents of the vendor dir backward compatible).

This package definition is a hack that lets the recent compilers use module mode to ensure that nothing is downloaded a build time.

> What actually prevents us from using the GoPackage base class for direnv?

The project builds itself with a Makefile, `MakefilePackage` seems like the right tool for the jobs.

> Does GoPackage need to be more flexible?

I don't think so.  Simple projects can use GoPackage.  Bigger projects that are driven by Makefiles should use `MakefilePackage`.